eng/intellisense.targets

```diff
@@ -0,0 +1,41 @@
+<Project>
+
+  <PropertyGroup>
+    <XmlDocFileRoot>$([MSBuild]::NormalizeDirectory('$(NuGetPackageRoot)', 'microsoft.private.intellisense', '$(MicrosoftPrivateIntellisenseVersion)', 'IntellisenseFiles'))</XmlDocFileRoot>
+  </PropertyGroup>
+
+  <!-- If the project is not generating its own documentation yet, retrieve it from the dotnet-api-docs nuget package. -->
+  <Choose>
+    <When Condition="('$(UseIntellisenseDocumentationFile)' == '' or '$(UseIntellisenseDocumentationFile)' == 'false') and Exists($([MSBuild]::NormalizePath('$(XmlDocFileRoot)', 'net', '1033', '$(AssemblyName).xml')))">
+      <PropertyGroup>
+        <IntellisenseFilePath>$([MSBuild]::NormalizePath('$(XmlDocFileRoot)', 'net', '1033', '$(AssemblyName).xml')</IntellisenseFilePath>
+      </PropertyGroup>
+    </When>
+    <When Condition="('$(UseIntellisenseDocumentationFile)' == '' or '$(UseIntellisenseDocumentationFile)' == 'false') and Exists($([MSBuild]::NormalizePath('$(XmlDocFileRoot)', 'dotnet-plat-ext', '1033', '$(AssemblyName).xml')))">
+      <PropertyGroup>
+        <IntellisenseFilePath Condition="">$([MSBuild]::NormalizePath('$(XmlDocFileRoot)', 'dotnet-plat-ext', '1033', '$(AssemblyName).xml')</IntellisenseFilePath>
+      </PropertyGroup>
+    </When>
+  </Choose>
+
+  <!-- TODO: Remove this target when no library relies on the intellisense documentation file anymore.-->
+  <Target Name="ChangeDocumentationFileForPackaging"
```

Comment:
As we just discussed, this might be too late to impact the result of a ProjectReference.

ProjectReferences (as used in the transport packages) only recieve the path to the dll in the bin folder and find the doc file relative to that.

To support  ProjectReference we may want to swap the value of `@(DocFileItem)` instead and do so right before `CopyFilesToOutputDirectory`.  This will make it so that the "shipping" xml is always next to the binary in the bin folder.  The compiler generated one (in the case it is not shipping) will remain in the obj folder.

Reply:
Here's a sample that does this: https://github.com/ericstj/runtime/commit/94849babba1fdcffec9a5a53fc0792377e0ae7fb

Reply:
Die you verify that this works in a pack-only scenario as well? dotnet pack --no-build

Reply:
Added some tests in a comment in the main thread.